mcs/class/corlib/Test/System.IO/FileTest.cs

```diff
@@ -2680,5 +2682,59 @@ public void MoveTest ()
 			MoveTest (FileAccess.ReadWrite, FileShare.Write | FileShare.Delete, true);
 			MoveTest (FileAccess.ReadWrite, FileShare.ReadWrite | FileShare.Delete, true);
 		}
+
+
+		[DllImport ("libc", SetLastError=true)]
+		public static extern int symlink (string oldpath, string newpath);
+
+		[DllImport ("libc", SetLastError=true)]
+		public static extern int rename (string oldpath, string newpath);
+
+		static void DeleteBadSymLink (string path) {
+			string newpath = path + new Random ().Next ();
```

Comment:
the `Random` instance should be shared, creating two instances in rapid succession like with the calls in the unit test most likely causes it to use the same seed.